src/main/java/org/eclipsefoundation/marketplace/model/QueryParams.java

```diff
@@ -112,4 +110,12 @@ public void unsetParam(String key) {
 		this.params.remove(key);
 	}
 
+	/**
+	 * Returns a copy of the internal map of parameters safely.
+	 * 
+	 * @return a copy of the internal param map
+	 */
+	public Map<String, List<String>> getInternalParamMap() {
```

Comment:
As this method leak internal representation, make it package-private

Reply:
If I follow the below comment of removing CachableResource, then this can't be package private. I can look at passing back param keys instead, and mapping the list of keys to a map by getting the values. That adds operations but is safer as it doesn't pass references of the direct map back. Does that work?

Reply:
My bad, did not notice the different package... Keep this method. However, I would name it `asMap` just to make it clear that it's not about the internal representation (even if the return value is a copy of the encapsulated map). The javadoc should say "returns this QueryParams object as a Map of param values indexed by the param name".